src/pages/sa-clients/forms/SalesForm.js

```diff
@@ -0,0 +1,172 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { useForm } from 'src/hooks/form'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { ControlContext } from 'src/providers/ControlContext'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { useInvalidate } from 'src/hooks/resource'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+
+export default function SalesForm({ labels, maxAccess, recordId, store }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: SaleRepository.ClientGroups.page
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      ...store.record
+    },
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      tdPct: yup
+        .number()
+        .max(99.9)
+        .nullable()
+        .transform((value, originalValue) => (originalValue === '' ? null : value))
+        .test(function (value) {
+          const { maxDiscount } = this.parent
+          if (value != null && maxDiscount != null) {
+            return value <= maxDiscount
+          }
+
+          return true
+        }),
+      maxDiscount: yup
+        .number()
+        .max(100)
+        .nullable()
+        .transform((value, originalValue) => (originalValue === '' ? null : value))
+    }),
+    onSubmit: async obj => {
+      const response = await postRequest({
+        extension: SaleRepository.Client.set,
+        record: JSON.stringify(obj)
+      })
+      if (!obj.recordId) {
+        toast.success(platformLabels.Added)
+        formik.setFieldValue('recordId', response.recordId)
+      } else toast.success(platformLabels.Edited)
+      invalidate()
+    }
```

Comment:
![image](https://github.com/user-attachments/assets/d8c803ad-febe-4b00-ab19-9d0a5d3d822f)
make sure there isn't any mandatory fields needed here
and of there is not check why this error is shown

Reply:
![image](https://github.com/user-attachments/assets/5d101700-bad9-47bb-beca-f4b8bd5e7dbc)
even after filling the fields